hibernate-core/src/main/java/org/hibernate/sql/results/graph/collection/internal/DelayedCollectionInitializer.java

```diff
@@ -85,7 +87,9 @@ public void resolveKey(RowProcessingState rowProcessingState) {
 						collectionAttributeMapping.getCollectionDescriptor(),
 						parentKey
 				);
-				parentAccess.registerResolutionListener( owner -> collectionInstance.setOwner( owner ) );
+				if ( collectionInstance != null ) {
```

Comment:
Again, shouldn't this if move into the lambda instead? The field is only initialized when `resolveInstance` is called.

Reply:
See my earlier comment.